src/sagas.js

```diff
@@ -485,8 +483,8 @@ function* getConsoleVm (action) {
  * VmDetail is to be rendered.
  */
 export function* selectVmDetail (action) {
-  yield put(setVmDetailToShow({ vmId: action.payload.vmId }))
   yield fetchSingleVm(getSingleVm({ vmId: action.payload.vmId })) // async data refresh
+  yield put(loadInProgress({ value: false }))
```

Comment:
According to what I see code, loadInProgress set only two times, first in begining when with reducer, and second time in login sagas, that's all, and in code it present only two times, in login sagas and fetchAllVms sagas, so I don't see any reasons why it can't be used for showing Loading component when page is loading.
There is react-router, so when page is reloaded, all previous data will be lost, and web-ui need to load this data again for certain page, until it is not done I think better to show Loading component for user.